### ISSUE TITLE
Whitespace removal with indentation

### ISSUE THREAD

**Issue by kentarohonda** [haml/haml#1114]
Hello.

Upgrading from 5.2.2 to 6.0.10, I find outer whitespace removal `>` doesn't remove some newlines.
A minimal example is the following.
```haml
- if true
  a
%br>
b
```
is compiled to
```html
a
<br>b
```

But, Haml 5.2.2 compiles it to
```html
a<br>b
```
, which is my expected output.

I produced this example using Haml 6.0.10
```
# haml version
6.0.10
```

Thank you.

**Comment by k0kubun** [comment]
Fixed in 6.0.11.

**Comment by kentarohonda** [comment]
@k0kubun 
Thank you for fixing.
I confirm v6.0.11 works well for the above example, but I find a newline still remains when else block exists and if block is rendered.
As a new minimal example,
```haml
- if true
  a
- else
  b
%br>
c
```
will be
```html
 a
<br>c
```
though the output of v5.2.2 is
```html
a<br>c
```

**Comment by k0kubun** [comment]
Fixed in v6.0.12.

**Comment by kentarohonda** [comment]
Thank you for so quick fixing!
